sbi/utils/plot/plot.py

```diff
@@ -0,0 +1,800 @@
+        # options for plot
+        "plot_offdiag": {},
+        # formatting points (scale, markers)
+        "points_diag": {},
+        "points_offdiag": {"marker": ".", "markersize": 20,},
+        # matplotlib style
+        "style": os.path.join(os.path.dirname(__file__), "matplotlibrc"),
+        # other options
+        "fig_size": (10, 10),
+        "fig_bg_colors": {"upper": None, "diag": None, "lower": None},
+        "fig_subplots_adjust": {"top": 0.9,},
+        "subplots": {},
+        "despine": {"offset": 5,},
+        "title_format": {"fontsize": 16},
+    }
+    # TODO: add color map support
+    # TODO: automatically determine good bin sizes for histograms
+    # TODO: add legend (if legend is True)
+
+    samples_nd.defaults = opts.copy()
+    opts = _update(opts, kwargs)
+
+    # Prepare samples
+    if type(samples) != list:
+        samples = [samples]
+
+    # Prepare points
+    if type(points) != list:
+        points = [points]
+    points = [np.atleast_2d(p) for p in points]
+
+    # Dimensions
+    dim = samples[0].shape[1]
+    num_samples = samples[0].shape[0]
+
+    # TODO: add asserts checking compatiblity of dimensions
+
+    # Prepare labels
+    if opts["labels"] == [] or opts["labels"] is None:
+        labels_dim = ["dim {}".format(i + 1) for i in range(dim)]
+    else:
+        labels_dim = opts["labels"]
+
+    # Prepare limits
+    if opts["limits"] == [] or opts["limits"] is None:
+        limits = []
+        for d in range(dim):
+            min = +np.inf
+            max = -np.inf
+            for sample in samples:
+                min_ = sample[:, d].min()
+                min = min_ if min_ < min else min
+                max_ = sample[:, d].max()
+                max = max_ if max_ > max else max
+            limits.append([min, max])
+    else:
+        if len(opts["limits"]) == 1:
+            limits = [opts["limits"][0] for _ in range(dim)]
+        else:
+            limits = opts["limits"]
+
+    # Prepare ticks
+    if opts["ticks"] == [] or opts["ticks"] is None:
+        ticks = None
+    else:
+        if len(opts["ticks"]) == 1:
+            ticks = [opts["ticks"][0] for _ in range(dim)]
+        else:
+            ticks = opts["ticks"]
+
+    # Prepare diag/upper/lower
+    if type(opts["diag"]) is not list:
+        opts["diag"] = [opts["diag"] for _ in range(len(samples))]
+    if type(opts["upper"]) is not list:
+        opts["upper"] = [opts["upper"] for _ in range(len(samples))]
+    # if type(opts['lower']) is not list:
+    #    opts['lower'] = [opts['lower'] for _ in range(len(samples))]
+    opts["lower"] = None
+
+    # Style
+    if opts["style"] in ["dark", "light"]:
+        style = os.path.join(
+            os.path.dirname(__file__), "matplotlib_{}.style".format(opts["style"])
+        )
+    else:
+        style = opts["style"]
+
+    # Apply custom style as context
+    with mpl.rc_context(fname=style):
```

Comment:
We could remove the mpl context here and the option -- it can be applied outside of the samples function when necessary